### ISSUE TITLE
.Net Framework 4.7.2 + AspNetCore 2.2 runtime exeption

### ISSUE THREAD

**Issue by gangstatracer** [andrewlock/NetEscapades.Configuration#36]
I have an issue using last NetEscapades.Configuration.Yaml along with Microsoft.Extensions.Configuration 2.2.0 in my project targeting .Net Framework 4.7.2 
Trying to build Configuration builder with yaml causes exception:

> Method not found: "System.Collections.Generic.Dictionary`2<System.String,System.Object> Microsoft.Extensions.Configuration.IConfigurationBuilder.get_Properties()".
at Microsoft.Extensions.Configuration.FileConfigurationExtensions.SetFileProvider(IConfigurationBuilder builder, IFileProvider fileProvider) ...

**Comment by andrewlock** [comment]
Hi, sorry for the delay. I've just tested this configuration but I don't get any errors. 

Is this still an issue, or did you figure out the problem?
Thanks

**Comment by gangstatracer** [comment]
Hi! I'm sorry, should have been provided more clear example. Here it is: https://github.com/gangstatracer/NetEscapades.Configuration.RuntimeIssueExample
**TL;DR**: key aspect of failng is
```xml
<PropertyGroup>
  <TargetFramework>net472</TargetFramework>
</PropertyGroup>
<ItemGroup>
  <PackageReference Include="NetEscapades.Configuration.Yaml" Version="1.6.0" />
  <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="2.2.0" />
</ItemGroup>
```

As you can see, using packet in library, that also uses `Microsoft.Extesions.Configuration.Abstractions 2.2.0` causes exception during `ConfigurationBuilder.Build()` invocation.

**Comment by gangstatracer** [comment]
Was able to fix by explicitly referencing
```xml
<PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="2.2.0" />
```

**Comment by andrewlock** [comment]
Ah, I see, this is part of a much bigger versioning problem in general. Currently the library is attempting to support ASP.NET Core 1.x as well as 2.x, and it's making some incorrect assumptions here: https://github.com/andrewlock/NetEscapades.Configuration/blob/23a57c6e7d7bbb2efc70c55c34173f4bae6d6c99/src/NetEscapades.Configuration.Yaml/NetEscapades.Configuration.Yaml.csproj#L27-L30

I'm inclined to remove the netstandard 1.3 support entirely, bump up the target framework support to 472, and remove the conditional versions. Clearly my attempt to be more inclusive just doesn't work properly 🙁

Glad you figured it out!

**Comment by gangstatracer** [comment]
Ok, waiting for new release and use temporary fix by now. Thanks for nice library!
